src/Microsoft.Health.Fhir.Api/Features/SMART/SmartClinicalScopesMiddleware.cs

```diff
@@ -45,38 +47,58 @@ public async Task Invoke(HttpContext context, RequestContextAccessor<IFhirReques
             {
                 var fhirRequestContext = fhirRequestContextAccessor.RequestContext;
                 var principal = fhirRequestContext.Principal;
-                fhirRequestContext.AccessControlContext.ApplyFineGrainedAccessControl = true;
+                var roles = principal.FindAll(authorizationConfiguration.RolesClaim).Select(r => r.Value);
 
-                // examine the scopes claim for any SMART on FHIR clinical scopes
-                DataActions permittedDataActions = 0;
-                foreach (Claim claim in principal.FindAll(authorizationConfiguration.ScopesClaim))
+                // Only read and apply SMART clinical scopes if the SmartUserRole is present
+                if (roles.Contains(authorizationConfiguration.SmartUserRole))
                 {
-                    var matches = ClinicalScopeRegEx.Matches(claim.Value);
-                    foreach (Match match in matches)
-                    {
-                        fhirRequestContext.AccessControlContext.ClinicalScopes.Add(match.Value);
+                    fhirRequestContext.AccessControlContext.ApplyFineGrainedAccessControl = true;
 
-                        var id = match.Groups["id"]?.Value;
-                        var resource = match.Groups["resource"]?.Value;
-                        var accessLevel = match.Groups["accessLevel"]?.Value;
+                    var fhirUser = principal.FindFirstValue(authorizationConfiguration.FhirUserClaim);
```

Comment:
May fhirUser be null?

Reply:
No, it is not allowed. If fhirUser claim is not present, it will return a null, and then we will throw a BadRequest exception, because if the user is in the SmartUser role, fhirUser claim is required.